test/junit/scala/ExtractorTest.scala

```diff
@@ -23,10 +23,11 @@ class ExtractorTest extends RunTesting {
     }
   }
 
-  @Test(expected = classOf[MatchError]) def testOptionalFunctionExtractorMatchError(): Unit = {
-    2 match {
+  @Test
+  def testOptionalFunctionExtractorMatchError(): Unit = {
+    assertThrows(classOf[MatchError], () => 2 match {
```

Comment:
there's some argument that `assertThrows` from `scala.tools.testkit.AssertUtil` would be a bit cleaner, but it's probably not super important

Reply:
definitely preferable to prefer scala API.